sbncode/CAFMaker/CAFMaker_module.cc

```diff
@@ -1280,11 +1278,17 @@ void CAFMaker::produce(art::Event& evt) noexcept {
         thisShower = fmShower.at(iPart);
       }
 
-      if (!thisTrack.empty())  { // it's a track!
+      assert(thisParticle.size() == 1);
+     
+      rec.reco.pfp.push_back(SRPFP());
```

Comment:
This element is referenced many, many times in this code.
I suggest one of the two:
1. create a local variable, fill it as needed, then add it to the collection at the end (my personal favourite):
    ```suggestion
          SRPFP pfp;
          // ...
          rec.reco.pfp.push_back(std::move(pfp));
    ```
2. reference this last element with some name and use that name from now on:
    ```suggestion
          rec.reco.pfp.push_back(SRPFP());
          SRPFP& pfp = rec.reco.pfp.back();
          // ...
    ```